addons/website/static/src/js/content/inject_dom.js

```diff
@@ -52,5 +67,26 @@ document.addEventListener('DOMContentLoaded', () => {
     }
     htmlEl.dataset.logged = !session.is_website_user;
 
+    document
+        .querySelectorAll(".o_mega_menu section.o_snippet_desktop_invisible")
+        .forEach((el) => el.closest("li").classList.add("d-none"));
+
+    // Since Mega Menus are located in the desktop header at first, we need
+    // to get the indices of the mega menu elements to hide the correct one
+    // in mobile
+    const megaMenuSectionEls = document.querySelectorAll(".o_mega_menu section");
+    const noMobileMegaMenuEls = document.querySelectorAll(
+        ".o_mega_menu section.o_snippet_mobile_invisible"
+    );
+    if (noMobileMegaMenuEls.length != 0) {
+        const mobileMegaMenuToggleEls = document.querySelectorAll(
```

Comment:
It gets all the `<li>` elements, not just the ones for the mega menu.
So the naming isn’t right, and it will also lead to errors.

Example that leads to an error:
Add two mega menus, hide one on mobile and the other on desktop.
Then save the page and test both mega menus on mobile. You’ll see that the only visible menu is the desktop one: only the button appears, while the mega menu itself is invisible.

I guess the same issue occurs for the "conditional visibility" code